README.md

```diff
@@ -6,247 +6,218 @@ Midtrans (formerly known as Veritrans Indonesia).
-2. Simplepay, demonstrated how to integrate Midtrans with a simple, succint Rails project. [Have a look](https://github.com/veritrans/veritrans-ruby/tree/master/example/rails/simplepay).
-3. Cable, demonstrate a chat-commerce app where the payment is handled with Midtrans. [Have a look](https://github.com/veritrans/veritrans-ruby/tree/master/example/rails/cable).
 
-## How to use (Rails)
+## How to use
 
 ### Add gem veritrans to Gemfile
 
 ```ruby
 gem 'veritrans'
+```
+```ruby
+gem install veritrans
 ```
 
     bundle install
 
-### Generate veritrans.yml
+### General setting
+Create instance of Midtrans client
 
-    rails g veritrans:install
+```ruby
+require 'veritrans'
+
+mt_client = Midtrans.new(
+    server_key: "your server key",
+    client_key: "your client key",
+    api_host: "https://api.sandbox.midtrans.com", # default
+    http_options: { }, # optional
+    logger: Logger.new(STDOUT), # optional
+    file_logger: Logger.new(STDOUT), # optional
+  )
+
+mt_client.status("order-id-123456")
+```
 
-### Create simple payment form (optional)
+also can set one by one
 
-    rails g veritrans:payment_form
+```ruby
+Midtrans.config.server_key = "your server key"
+Midtrans.config.client_key = "your client key"
+Midtrans.config.api_host = "https://api.sandbox.midtrans.com"
+```
 
-### Edit api keys in config/veritrans.yml
+### Override Notification URL
 
-```yml
-# config/veritrans.yml
-development:
-  client_key: # your api client key
-  server_key: # your api client key
+You can opt to change or add custom notification urls on every transaction. It can be achieved by adding additional HTTP headers into charge request.
+```ruby
+# Add new notification url(s) alongside the settings on Midtrans Dashboard Portal (MAP)
+$append_notif_url = "https://example.com/test1,https://example.com/test2"
+# Use new notification url(s) disregarding the settings on Midtrans Dashboard Portal (MAP)
+$override_notif_url = "https://example.com/test1"
 ```
 
-## STEP 1: Process credit cards
+[More details](https://api-docs.midtrans.com/#override-notification-url)
+> **Note:** When both `appendNotifUrl` and `overrideNotifUrl` are used together then only `overrideNotifUrl` will be used.
 
-#### Snap
+> Both header can only receive up to maximum of **3 urls**.
 
-##### Pop-up
+### Idempotency-Key
+You can opt to add idempotency key on charge transaction. It can be achieved by adding additional HTTP headers into charge request.
+Is a unique value that is put on header on API request. Midtrans API accept Idempotency-Key on header to safely handle retry request
+without performing the same operation twice. This is helpful for cases where merchant didn't receive the response because of network issue or other unexpected error.
+```ruby
+$idempotency_key = "Unique-ID"
+```
+[More details](http://api-docs.midtrans.com/#idempotent-requests)
+
+
+## STEP 1: Process credit cards
```

Comment:
The old structure of this readme is no longer relevant (why is credit card step 1? why non card step 2? doesn't make sense), can you restructure like the other library we have update (e.g https://github.com/Midtrans/midtrans-nodejs-client#2-usage). So it is not step 1, step 2, but reader choose which product Snap, Core API, etc.